### DISCUSSION TITLE
How to run mitmproxy v7 on Python3 and Raspberry Pi?

### DISCUSSION THREAD

**Discussion by ghostlexly** [mitmproxy/mitmproxy#5036] [Q&A] [answered]
Hello, i tryed pip3 install mitmproxy and it's can only find the v5.3.0.
How can i install the v7 on Raspberry Pi 4 ?

**Comment by mhils** [comment]
Presumably you need a newer Python version. v7 requires Python 3.8. Alternatively you can look at our precompiled ARM Docker Images. :)

**Comment by ghostlexly** [comment]
I have Python 3.10 installed

i have the same problem with the docker image:
`ERROR: no matching manifest for linux/arm/v7 in the manifest list entries`

**Reply by mhils** [reply, marked as answer]
Right - we only provide arm64 images at the moment as cryptograpy fails to compile on armv7 (https://github.com/pyca/cryptography/issues/6347). You'd need a 64 bit OS on your Raspberry Pi. Alternatively you can probably install cryptography from piwheels.org :)
